api/server.py

```diff
@@ -26,31 +26,58 @@ def run(port, hostname, debug_mode):
     app.run(host=hostname, port=port, debug=debug_mode)
 
 
-# TODO: If meta-paths for A and B will be written in Java, they will need this information in Java
+# TODO: If functionality "meta-paths for node set A and B" will be written in Java, team alpha will need this information in Java
 @app.route("/node-sets", methods=["POST"])
 def receive_node_sets():
+    """
+    Endpoint where the node sets from the "Setup" page which the user selects are posted.
+    This endpoint is called for each new added node.
+
+    The repeated calling enables us to start the following computations as early as possible
+    so that we can return information on the next pages faster.
+    For example on the first call we already know the type of the whole node set and
+    therefore can begin to retrieve the corresponding node sets.
+    """
     # TODO: Check if necessary information is in request object
     if not request.json:
         abort(400)
+    raise NotImplementedError("This API endpoint isn't implemented in the moment")
 
 
 @app.route("/node-sets", methods=["GET"])
 def send_node_sets():
+    """
+    Endpoint where the node sets which the user previously selected on the "Setup" page can be retrieved.
```

Comment:
"Endpoint where..." sounds a bit weird. Maybe just "Returns..." or "Receives..." depending on the method and leave out *Endpoint* as it should be clear that these are such. This applies to all comments.